### PR TITLE
Integrate FXA for User Accounts #28

### PR DESCRIPTION
**Pull request by jaredkerim**
- Add FXA client
- Add FXA post registration redirect view
- Tests

### CODE REVIEW

**Comment by jaredkerim** [comment]
@crankycoder r?

File: leaderboard/fxa/client.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    def _parse_response(self, response):
+        """
+        Ensure that a response returned status 200 then parse
+        and return its JSON contents.
+        """
+        if response.status_code != 200:
+            raise FXAException('Request Error: {}'.format(response.content))
+
+        try:
+            json_data = json.loads(response.content)
```
Comment (crankycoder) [inline leaderboard/fxa/client.py]: requests implements response.json()

http://docs.python-requests.org/en/latest/api/#requests.Response.json

File: leaderboard/fxa/urls.py [hunk 2]
```diff
@@ -0,0 +1,8 @@
+from django.conf.urls import url
+
+from leaderboard.fxa.views import FXARedirectView
+
+urlpatterns = [
+    url('^redirect/', FXARedirectView.as_view(),
+        name='fxa-redirect'),
```
Comment (crankycoder) [inline leaderboard/fxa/urls.py]: I would pull out 'fxa-redirect' as a constant as you're referencing it in reverse() function calls.

File: leaderboard/fxa/views.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+    def get(self, request):
+        code = request.GET.get('code', '')
+
+        if not code:
+            raise ValidationError('Unable to determine access code.')
+
+        try:
+            token_data = self.fxa_client.get_authorization_token(code)
+        except FXAException:
```
Comment (crankycoder) [inline leaderboard/fxa/views.py]: I think this won't catch network I/O errors if the FxA server is down.

**Pull request merged**